unit_tests/test_socket_connection.py

```diff
@@ -595,40 +604,51 @@ def test_raw_l3(self):
             b'"And pray where in earth or heaven are there any prudent marriages?""'
         )
 
+        data_to_receive = b"Procuring education on consulted assurance in do."
+
         # Given
         server = MiniTestServer(proto="raw", host="lo")
-        server.data_to_send = "GKC"
         server.bind()
 
+        # noinspection PyDeprecation
         uut = SocketConnection(host="lo", proto="raw-l3")
         uut.logger = logging.getLogger("SulleyUTLogger")
 
         # Assemble packet...
-        raw_packet = ip_packet(
+        raw_packet_send = ip_packet(
             payload=udp_packet(payload=data_to_send, src_port=server.active_port + 1, dst_port=server.active_port),
             src_ip=b"\x7F\x00\x00\x01",
             dst_ip=b"\x7F\x00\x00\x01",
         )
-        expected_server_receive = b"\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00" + raw_packet
+        expected_server_receive = b"\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00" + raw_packet_send
+
+        expected_client_receive = ip_packet(
+            payload=udp_packet(payload=data_to_receive, src_port=server.active_port, dst_port=server.active_port + 1),
+            src_ip=b"\x7F\x00\x00\x01",
+            dst_ip=b"\x7F\x00\x00\x01",
+        )
+        server.data_to_send = b"\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00" + expected_client_receive
 
         t = threading.Thread(target=functools.partial(server.receive_until, expected_server_receive))
         t.daemon = True
         t.start()
 
         # When
         uut.open()
-        send_result = uut.send(data=raw_packet)
+        send_result = uut.send(data=raw_packet_send)
+        _ = uut.recv(10000)  # Receiving the sent package. Will not work on a busy socket!
```

Comment:
This seems to be a common problem with raw sockets.
Could be fixed by filtering received packages by their MAC-address.

Reply:
Will be dealt with in another PR.